addon/ExperimentProvider.js

```diff
@@ -1,32 +1,60 @@
 const seedrandom = require("addon/vendor.bundle").SeedRandom;
+const simplePrefs = require("sdk/simple-prefs");
+const OVERRIDE_PREF = "experimentOverrides";
 
+exports.OVERRIDE_PREF = OVERRIDE_PREF;
 exports.ExperimentProvider = class ExperimentProvider {
-  constructor(clientID, experiments = require("experiments.json"), rng) {
+  constructor(clientID, experiments = require("../experiments.json"), rng) {
     this._clientID = clientID;
+    this._experiments = experiments;
     this._rng = rng || seedrandom(clientID);
     this._data = {};
     this._experimentId = null;
+  }
+
+  init() {
+    this.setValues();
+    this._onPrefChange = () => this.setValues();
+    simplePrefs.on(OVERRIDE_PREF, this._onPrefChange);
+  }
+
+  setValues() {
+    // Load override pref, if defined
+    const overrides = simplePrefs.prefs[OVERRIDE_PREF] && simplePrefs.prefs[OVERRIDE_PREF].split(/,?\s+/);
 
     const randomNumber = this._rng();
     let floor = 0;
 
-    Object.keys(experiments).forEach(key => {
-      const experiment = experiments[key];
-      if (experiment.active === false) {
+    Object.keys(this._experiments).forEach(key => {
+      let inExperiment;
+      const experiment = this._experiments[key];
+      if (!overrides && experiment.active === false) {
         return;
       }
       const {variant, control} = experiment;
-      const ceiling = variant.threshold + floor;
-      if (ceiling > 1) {
-        throw new Error("Your variant cohort sizes should add up to less than 1.");
-      }
-      const inExperiment = randomNumber >= floor && randomNumber < ceiling;
-      if (inExperiment) {
-        this._experimentId = variant.id;
+
+      // If an override pref is defined, use it to determine which
+      // experiment values are used. If not, just use our random number generator.
+      // Note that no _experimentId is set for manually overridden prefs.
+      if (overrides) {
+        inExperiment = overrides.includes(key);
+      } else {
+        const ceiling = variant.threshold + floor;
+        if (ceiling > 1) {
+          throw new Error("Your variant cohort sizes should add up to less than 1.");
+        }
+        inExperiment = randomNumber >= floor && randomNumber < ceiling;
+        if (inExperiment) {
+          this._experimentId = variant.id;
```

Comment:
What if multiple experiments are active and user gets > 1? Won't this get set to the variant of the last experiment and lose previous information?